app/Database/MigrationCreator.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+/**
+ * @author    MyBB Group
+ * @version   2.0.0
+ * @package   mybb/core
+ * @license   http://www.mybb.com/licenses/bsd3 BSD-3
+ */
+
+namespace MyBB\Core\Database;
+
+use MyBB\Core\Extension\Extension;
+use Illuminate\Filesystem\Filesystem;
+
+class MigrationCreator
+{
+    /**
+     * The filesystem instance.
+     *
+     * @var Filesystem
+     */
+    protected $files;
+
+    /**
+     * @var string
+     */
+    protected $publicPath;
+
+    /**
+     * Create a new migrator instance.
+     *
+     * @param Filesystem $files
+     * @param string $publicPath
+     */
+    public function __construct(Filesystem $files, $publicPath)
```

Comment:
We're requiring PHP7, so feel free to use scalar type hints - here for instance `string $publicPath`.